Assignment4/s100257_SPARQL(Basic).txt

```diff
@@ -0,0 +1,311 @@
+|	http://GP-onto.fi.upm.es/exercise2#SecurityService          |	SecurityService         |
+|	http://GP-onto.fi.upm.es/exercise2#Track                    |	Track                   |
+|	http://GP-onto.fi.upm.es/exercise2#Train                    |	Train                   |
+|	http://GP-onto.fi.upm.es/exercise2#Railway                  |	Railway                 |
+|	http://GP-onto.fi.upm.es/exercise2#Village                  |	Village                 |
++-------------------------------------------------------------+-------------------------+
+
+
+2. Get all the subclasses of the class Establishment
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?SubClass ?Label
+WHERE {
+	?SubClass rdfs:subClassOf onto:Establishment.
+	?SubClass rdfs:label ?Label.
+}
+
++-----------------------------------------------+-------------+
+|	SubClass                                      |	Label       |
++-----------------------------------------------+-------------+
+|	http://GP-onto.fi.upm.es/exercise2#Hostel     |	Hostel      |
+|	http://GP-onto.fi.upm.es/exercise2#GuestHouse |	GuestHouse  |
+|	http://GP-onto.fi.upm.es/exercise2#Hotel      |	Hotel       |
++-----------------------------------------------+-------------+
+
+
+3. Get all instances of the class City
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?Instance ?Label
+WHERE {
+	?Instance a onto:City.
+	?Instance rdfs:label ?Label.
+}
+
++-----------------------------------------------------------+-------------------------+
+|	Instance                                                  |	Label                   |
++-----------------------------------------------------------+-------------------------+
+|	http://GP-onto.fi.upm.es/exercise2#ACoruC1a               |	A Coruña                |
+|	http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela |	Santiago_de_Compostela  |
+|	http://GP-onto.fi.upm.es/exercise2#Madrid                 |	Madrid                  |
++-----------------------------------------------------------+-------------------------+
+
+
+4. Get the number of inhabitants of Santiago de Compostela
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?InhabitantNumber
+WHERE {
+	onto:Santiago_de_Compostela onto:hasInhabitantNumber ?InhabitantNumber.
+}
+
++---------------------------------------------------------+
+|	InhabitantNumber                                        |
++---------------------------------------------------------+
+|	" 300000 "^^<http://www.w3.org/2001/XMLSchema#integer>  |
++---------------------------------------------------------+
+
+
+5. Get the number of inhabitants of Santiago de Compostela and Arzua
+
+PREFIX onto: <http://GP-onto.fi.upm.es/exercise2#>
+
+SELECT DISTINCT
+	?Place ?Label ?InhabitantNumber
+WHERE {
+	{
+		?Place onto:hasInhabitantNumber ?InhabitantNumber.
+		filter (?Place = onto:Santiago_de_Compostela).
```

Comment:
You do not need to separate into two lines, but can just put onto:Santiago_de_Compostela instead of ?Place. Same below.